<!DOCTYPE html>
<html lang="en" prefix="og: http://ogp.me/ns# fb: https://www.facebook.com/2008/fbml">
<head>
    <title>pdb to fasta conversion with Bash - Plouf</title>
    <!-- Using the latest rendering mode for IE -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">


    <link href="/favicon.ico" rel="icon">

<link rel="canonical" href="/pdb2fasta/">

        <meta name="author" content="Pierre" />
        <meta name="keywords" content="fasta,pdb,structure,en,Unix/Linux" />
        <meta name="description" content="Here is a simple pdb to fasta format conversion tool. The purpose was to extract residue sequence from atomic coordinates (ATOM lines). Note that the sequence extracted this way corresponds to residues which are really present in the 3D structure. This could be different from the sequence obtained from the …" />

        <meta property="og:site_name" content="Plouf" />
        <meta property="og:type" content="article"/>
        <meta property="og:title" content="pdb to fasta conversion with Bash"/>
        <meta property="og:url" content="/pdb2fasta/"/>
        <meta property="og:description" content="Here is a simple pdb to fasta format conversion tool. The purpose was to extract residue sequence from atomic coordinates (ATOM lines). Note that the sequence extracted this way corresponds to residues which are really present in the 3D structure. This could be different from the sequence obtained from the …"/>
        <meta property="article:published_time" content="2012-03-14" />
            <meta property="article:section" content="blog" />
            <meta property="article:tag" content="fasta" />
            <meta property="article:tag" content="pdb" />
            <meta property="article:tag" content="structure" />
            <meta property="article:tag" content="en" />
            <meta property="article:tag" content="Unix/Linux" />
            <meta property="article:author" content="Pierre" />



    <!-- Bootstrap -->
        <link rel="stylesheet" href="/theme/css/bootstrap.simplex.min.css" type="text/css"/>
    <link href="/theme/css/font-awesome.min.css" rel="stylesheet">

    <link href="/theme/css/pygments/default.css" rel="stylesheet">
    <link href="/theme/tipuesearch/tipuesearch.css" rel="stylesheet">
    <link rel="stylesheet" href="/theme/css/style.css" type="text/css"/>
        <link href="/static/custom.css" rel="stylesheet">


        <link href="/feeds/all.rss.xml" type="application/rss+xml" rel="alternate"
              title="Plouf RSS Feed"/>

</head>
<body>

<div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-ex1-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a href="/" class="navbar-brand">
Plouf            </a>
        </div>
        <div class="collapse navbar-collapse navbar-ex1-collapse">
            <ul class="nav navbar-nav">
                    <li><a href="/">Home</a></li>
                    <li><a href="/about/">About</a></li>
                    <li><a href="/research/">Research</a></li>
                    <li><a href="/python/">Python</a></li>
                    <li><a href="/tags.html">Tags</a></li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
              <li><span>
                <form class="navbar-search" action="/search.html">
                  <input type="text" class="search-query" placeholder="Search" name="q" id="tipue_search_input" required>
                </form></span>
              </li>
            </ul>
        </div>
        <!-- /.navbar-collapse -->
    </div>
</div> <!-- /.navbar -->

<!-- Banner -->
<!-- End Banner -->

<!-- Content Container -->
<div class="container">
    <div class="row">
        <div class="col-sm-9">
    <section id="content">
        <article>
            <header class="page-header">
                <h1>
                    <a href="/pdb2fasta/"
                       rel="bookmark"
                       title="Permalink to pdb to fasta conversion with Bash">
                        pdb to fasta conversion with Bash
                    </a>
                </h1>
            </header>
            <div class="entry-content">
                <div class="panel">
                    <div class="panel-body">
<footer class="post-info">
    <span class="label label-default">Date</span>
    <span class="published">
        <i class="fa fa-calendar"></i><time datetime="2012-03-14T14:53:00+01:00"> mer. 14 mars 2012</time>
    </span>
          <span class="label label-default">Modified</span>
            <span class="modified">
                <i class="fa fa-calendar"></i><time datetime="2020-02-21T00:00:00+01:00"> ven. 21 février 2020</time>
            </span>





<span class="label label-default">Tags</span>
	<a href="/tag/fasta/">fasta</a>
        /
	<a href="/tag/pdb/">pdb</a>
        /
	<a href="/tag/structure/">structure</a>
        /
	<a href="/tag/en/">en</a>
        /
	<a href="/tag/unixlinux/">Unix/Linux</a>
    
</footer><!-- /.post-info -->                    </div>
                </div>
                <p>Here is a simple
<a href="http://www.wwpdb.org/documentation/format33/v3.3.html" title="PDB format">pdb</a>
to <a href="http://en.wikipedia.org/wiki/FASTA_format" title="fasta format">fasta</a>
format conversion tool. The purpose was to extract residue sequence from
atomic coordinates (<code>ATOM</code> lines).</p>
<p>Note that the sequence extracted this way corresponds to
residues which are really present in the 3D structure. This could be
different from the sequence obtained from the
<a href="http://www.wwpdb.org/documentation/format33/sect3.html#SEQRES" title="PDB - SEQRES documentation">SEQRES</a>
field -- protein sequence without missing residues due to experiment --
or from the
<a href="http://www.wwpdb.org/documentation/format33/sect3.html#DBREF" title="PDB - DBREF documentation">DBREF</a> field
-- complete sequence of the protein, provided through an UniProt
identifier (see final remark <a href="#PS2">PS 2</a> for an example).</p>
<p>This could have been done very quickly with a Python script but the goal
was also to practise Unix tools such as awk, sed, tr, fold... (grep was
not used here).</p>
<p>In the PDB format, the region of interest for this project are lines
starting with the <code>ATOM</code> keyword. Below is an excerpt from the
<a href="http://www.rcsb.org/pdb/explore.do?structureId=1BTA" title="1BTA structure">1BTA</a> structure
of Barstar.</p>
<div class="highlight"><pre><span></span><code><span class="o">[</span>...<span class="o">]</span>
ATOM     <span class="m">48</span>  CA  ALA A   <span class="m">3</span>      -1.086   <span class="m">3</span>.403   <span class="m">6</span>.848  <span class="m">1</span>.00  <span class="m">0</span>.30           C
ATOM     <span class="m">49</span>  C   ALA A   <span class="m">3</span>       <span class="m">0</span>.316   <span class="m">4</span>.013   <span class="m">6</span>.751  <span class="m">1</span>.00  <span class="m">0</span>.30           C
ATOM     <span class="m">50</span>  O   ALA A   <span class="m">3</span>       <span class="m">0</span>.469   <span class="m">5</span>.207   <span class="m">6</span>.586  <span class="m">1</span>.00  <span class="m">0</span>.54           O
ATOM     <span class="m">51</span>  CB  ALA A   <span class="m">3</span>      -1.491   <span class="m">2</span>.834   <span class="m">5</span>.486  <span class="m">1</span>.00  <span class="m">0</span>.34           C
ATOM     <span class="m">52</span>  H   ALA A   <span class="m">3</span>      -1.991   <span class="m">5</span>.353   <span class="m">6</span>.827  <span class="m">1</span>.00  <span class="m">0</span>.25           H
ATOM     <span class="m">53</span>  HA  ALA A   <span class="m">3</span>      -1.088   <span class="m">2</span>.618   <span class="m">7</span>.589  <span class="m">1</span>.00  <span class="m">0</span>.34           H
ATOM     <span class="m">54</span>  HB1 ALA A   <span class="m">3</span>      -2.257   <span class="m">3</span>.457   <span class="m">5</span>.050  <span class="m">1</span>.00  <span class="m">1</span>.08           H
ATOM     <span class="m">55</span>  HB2 ALA A   <span class="m">3</span>      -0.630   <span class="m">2</span>.813   <span class="m">4</span>.834  <span class="m">1</span>.00  <span class="m">1</span>.05           H
ATOM     <span class="m">56</span>  HB3 ALA A   <span class="m">3</span>      -1.871   <span class="m">1</span>.831   <span class="m">5</span>.612  <span class="m">1</span>.00  <span class="m">1</span>.07           H
ATOM     <span class="m">57</span>  N   VAL A   <span class="m">4</span>       <span class="m">1</span>.342   <span class="m">3</span>.214   <span class="m">6</span>.860  <span class="m">1</span>.00  <span class="m">0</span>.30           N
ATOM     <span class="m">58</span>  CA  VAL A   <span class="m">4</span>       <span class="m">2</span>.725   <span class="m">3</span>.769   <span class="m">6</span>.780  <span class="m">1</span>.00  <span class="m">0</span>.27           C
<span class="o">[</span>...<span class="o">]</span>
</code></pre></div>


<p>The 3rd, 4th and 5th columns correspond to the atom name, residue name
(3-letters code) and chain name, respectively. For a given protein
chain, we want the name of all residues in the structure. They all have
a <a href="http://en.wikipedia.org/wiki/Amino_acid" title="Wikipedia: amino acid">carbon alpha
atom</a>
labelled <code>CA</code> (provided no atom is missing). Therefore, we have to
select lines that start with the keyword <code>ATOM</code>, with column 3 equals to
<code>CA</code> and column 5 equals to the targeted chain (here chain <code>A</code>).</p>
<div class="highlight"><pre><span></span><code>$ cat 1BTA.pdb <span class="p">|</span> awk <span class="s1">&#39;/^ATOM/ &amp;&amp; $3 == &quot;CA&quot; &amp;&amp; $5 == &quot;A&quot; {print $4}&#39;</span>
</code></pre></div>


<p>We then get the sequence with one residue per line</p>
<div class="highlight"><pre><span></span><code><span class="err">LYS</span>
<span class="err">LYS</span>
<span class="err">ALA</span>
<span class="err">VAL</span>
<span class="err">ILE</span>
<span class="err">ASN</span>
<span class="err">[...]</span>
</code></pre></div>


<p>To remove the newline character after each residue, there are several
options:</p>
<ul>
<li><code>tr '\n' ' '</code> is easy to understand -- replace the newline ('\n')
    character by a space (' ') character</li>
<li><code>paste -s -d ' '</code> is more tricky</li>
<li>and <code>sed ':a;N;$!ba;s/\n/ /g'</code> is black magic to me (for an
    explanation, read this <a href="http://stackoverflow.com/questions/1251999/sed-how-can-i-replace-a-newline-n">nice
    thread</a>
    on StackOverflow)</li>
</ul>
<p>And from this</p>
<div class="highlight"><pre><span></span><code>$ cat 1BTA.pdb <span class="p">|</span> awk <span class="s1">&#39;/ATOM/ &amp;&amp; $3 == &quot;CA&quot; &amp;&amp; $5 == &quot;A&quot; {print $4}&#39;</span> <span class="p">|</span> tr <span class="s1">&#39;\n&#39;</span> <span class="s1">&#39; &#39;</span>
</code></pre></div>


<p>we get that</p>
<div class="highlight"><pre><span></span><code><span class="err">LYS LYS ALA VAL ILE ASN GLY GLU GLN ILE ARG SER ILE...</span>
</code></pre></div>


<p>Next we want to convert the 3-letter code residue name into 1-letter
code. <code>sed</code> is useful here</p>
<div class="highlight"><pre><span></span><code>$ sed <span class="s1">&#39;s/ALA/A/g;s/CYS/C/g;s/ASP/D/g;s/GLU/E/g;s/PHE/F/g;s/GLY/G/g;s/HIS/H/g;s/ILE/I/g;s/LYS/K/g;s/LEU/L/g;s/MET/M/g;s/ASN/N/g;s/PRO/P/g;s/GLN/Q/g;s/ARG/R/g;s/SER/S/g;s/THR/T/g;s/VAL/V/g;s/TRP/W/g;s/TYR/Y/g&#39;</span>
</code></pre></div>


<p>This is a pretty long command line since we have to convert the 20 amino
acids.</p>
<p>Eventually, we have to remove space between residue names with
<code>sed 's/ //g'</code> and fold the sequence into 60-character lines with
<code>fold -w 60</code>. The complete command line is</p>
<div class="highlight"><pre><span></span><code>$ cat 1BTA.pdb <span class="p">|</span> awk <span class="s1">&#39;/ATOM/ &amp;&amp; $3 == &quot;CA&quot; &amp;&amp; $5 == &quot;A&quot; {print $4}&#39;</span> <span class="p">|</span> tr <span class="s1">&#39;\n&#39;</span> <span class="s1">&#39; &#39;</span> <span class="p">|</span> sed <span class="s1">&#39;s/ALA/A/g;s/CYS/C/g;s/ASP/D/g;s/GLU/E/g;s/PHE/F/g;s/GLY/G/g;s/HIS/H/g;s/ILE/I/g;s/LYS/K/g;s/LEU/L/g;s/MET/M/g;s/ASN/N/g;s/PRO/P/g;s/GLN/Q/g;s/ARG/R/g;s/SER/S/g;s/THR/T/g;s/VAL/V/g;s/TRP/W/g;s/TYR/Y/g&#39;</span> <span class="p">|</span> sed <span class="s1">&#39;s/ //g&#39;</span> <span class="p">|</span> fold -w <span class="m">60</span>
</code></pre></div>


<p>that outputs</p>
<div class="highlight"><pre><span></span><code><span class="err">KKAVINGEQIRSISDLHQTLKKELALPEYYGENLDALWDCLTGWVEYPLVLEWRQFEQSK</span>
<span class="err">QLTENGAESVLQVFREAKAEGCDITIILS</span>
</code></pre></div>


<p>A more exhaustive script that scans all possible chains is available in
the
<a href="https://bitbucket.org/pierrepo/pdb2fasta" title="pdb2fasta repository (Bitbucket)">pdb2fasta</a>
repository from Bitbucket. You can get it with</p>
<div class="highlight"><pre><span></span><code>$ git clone https://bitbucket.org/pierrepo/pdb2fasta.git
</code></pre></div>


<p>or directly through the <a href="https://bitbucket.org/pierrepo/pdb2fasta/src" title="Bitbucket / pdb2fasta - source code brower">source code
browser</a>.
The repository also contains two PDB files in the <code>test</code> directory.</p>
<div class="highlight"><pre><span></span><code>$ ./pdb2fasta.sh test/1BTA.pdb
&gt;test/1BTA <span class="p">|</span> chain A <span class="p">|</span> <span class="m">89</span> aa
KKAVINGEQIRSISDLHQTLKKELALPEYYGENLDALWDCLTGWVEYPLVLEWRQFEQSK
QLTENGAESVLQVFREAKAEGCDITIILS
</code></pre></div>


<p>and</p>
<div class="highlight"><pre><span></span><code>$ ./pdb2fasta.sh test/3FCS.pdb
&gt;test/3FCS <span class="p">|</span> chain A <span class="p">|</span> <span class="m">934</span> aa
LNLDPVQLTFYAGPNGSQFGFSLDFHKDSHGRVAIVVGAPRTLGPSQEETGGVFLCPWRA
<span class="o">[</span>...<span class="o">]</span>
LKVDWDPVLVSCDSAPCTVVQCDLQEMARGQRAMVTVLAFLWLPSLYQRPLDQFVLQSHA
AWBWFNVSSLPYAVPPLSLPRGEAQVWTQLLRAC
&gt;test/3FCS <span class="p">|</span> chain B <span class="p">|</span> <span class="m">683</span> aa
GPNICTTRGVSSCQQCLAVSPMCAWCSDEALPLGSPRCDLKENLLKDNCAPESIEFPVSE
<span class="o">[</span>...<span class="o">]</span>
CTFKKECVECKKFDRGALHDENTCNRYCRDEIESVKELKDTGKDAVNCTYKNEDDCVVRF
QYYEDSSGKSILYVVEEPECCKG
&gt;test/3FCS <span class="p">|</span> chain C <span class="p">|</span> <span class="m">927</span> aa
LNLDPVQLTFYAGPNGSQFGFSLDFHKDSHGRVAIVVGAPRTLGPSQEETGGVFLCPWRA
<span class="o">[</span>...<span class="o">]</span>
KDPVLVSCDSAPCTVVQCDLQEMARGQRAMVTVLAFLWLPSLYQRPLDQFVLQSHAWFNV
SSLPYAVPPLSLPRGEAQVWTQLLRAC
&gt;test/3FCS <span class="p">|</span> chain D <span class="p">|</span> <span class="m">612</span> aa
GPNICTTRGVSSCQQCLAVSPMCAWCSDEALPLGSPRCDLKENLLKDNCAPAEBESIEFP
<span class="o">[</span>...<span class="o">]</span>
QCSCGDCLCDSDWTGYYCNCTTRTDTCMSSNGLLCSGRGKCECGSCVCIQPGSYGDTCEK
CPTCPDACTFKK
</code></pre></div>


<p>Have fun.</p>
<p>PS 1: beware that, in a PDB file, extracting data from the <code>ATOM</code> lines
using column numbers (the <code>awk</code> command line above) is not recommended.
Extracting a range of characters directly from the <code>ATOM</code> line is a much
more robust alternative. Although such manipulation is trivial with
Python, I had no idea how to do this in Bash  :-)</p>
<p>PS 2: to illustrate differences between the sequence
pointed in DBREF, contained in SEQRES and contained in the real
structure, the structure of chain B of PDB
<a href="http://www.rcsb.org/pdb/files/3FCS.pdb" title="PDB - 3FCS">3FCS</a> has 683
residues resolved experimentally, the sequence from SEQRES has 690
residues and the corresponding UniProt sequence (from DBREF) counts 788
residues.</p>
            </div>
            <!-- /.entry-content -->
<section class="well" id="related-posts">
    <h4>Related Posts:</h4>
    <ul>
        <li><a href="/pdb-format/">PDB format</a></li>
        <li><a href="/ambient-occlusion-pymol/">Ambient occlusion-like with PyMOL</a></li>
        <li><a href="/find-xargs-powerful-combo/">Find + xargs = powerful combo</a></li>
        <li><a href="/reduce-pdf-size/">Reduce PDF size</a></li>
        <li><a href="/cours-docking-gros-grain/">Cours docking gros grain [fr]</a></li>
    </ul>
</section>
        </article>
    </section>

        </div>
        <div class="col-sm-3" id="sidebar">
            <aside>
<!-- Sidebar -->
<section class="well well-sm">
  <ul class="list-group list-group-flush">

<!-- Sidebar/Social -->
<li class="list-group-item">
  <h4><i class="fa fa-home fa-lg"></i><span class="icon-label">Social</span></h4>
  <ul class="list-group" id="social">
    <li class="list-group-item"><a href="https://twitter.com/pierrepo"><i class="fa fa-twitter-square fa-lg"></i> Twitter</a></li>
    <li class="list-group-item"><a href="http://fr.linkedin.com/in/pierrepo"><i class="fa fa-linkedin-square fa-lg"></i> LinkedIn</a></li>
    <li class="list-group-item"><a href="https://github.com/pierrepo"><i class="fa fa-github-square fa-lg"></i> github</a></li>
    <li class="list-group-item"><a href="http://cupnet.net/feed/all.rss.xml"><i class="fa fa-rss-square fa-lg"></i> rss</a></li>
  </ul>
</li>
<!-- End Sidebar/Social -->
  </ul>
</section>
<!-- End Sidebar -->            </aside>
        </div>
    </div>
</div>
<!-- End Content Container -->

<footer>
   <div class="container">
      <hr>
      <div class="row">
         <div class="col-xs-10">&copy; 2020 Pierre Poulain
            &middot; Powered by <a href="https://github.com/getpelican/pelican-themes/tree/master/pelican-bootstrap3" target="_blank">pelican-bootstrap3</a>,
            <a href="http://docs.getpelican.com/" target="_blank">Pelican</a>,
            <a href="http://getbootstrap.com" target="_blank">Bootstrap</a>                <p><small>  <a rel="license" href="https://creativecommons.org/licenses/by/4.0/deed.en"><img alt="Creative Commons License" style="border-width:0" src="//i.creativecommons.org/l/by/4.0/80x15.png" /></a>
    Content
  licensed under a <a rel="license" href="https://creativecommons.org/licenses/by/4.0/deed.en">Creative Commons Attribution 4.0 International License</a>, except where indicated otherwise.
</small></p>
         </div>
         <div class="col-xs-2"><p class="pull-right"><i class="fa fa-arrow-up"></i> <a href="#">Back to top</a></p></div>
      </div>
   </div>
</footer>
<script src="/theme/js/jquery.min.js"></script>

<!-- Include all compiled plugins (below), or include individual files as needed -->
<script src="/theme/js/bootstrap.min.js"></script>

<!-- Enable responsive features in IE8 with Respond.js (https://github.com/scottjehl/Respond) -->
<script src="/theme/js/respond.min.js"></script>




</body>
</html>