<!DOCTYPE html>
<html lang="en" prefix="og: http://ogp.me/ns# fb: https://www.facebook.com/2008/fbml">
<head>
    <title>CV - Plouf</title>
    <!-- Using the latest rendering mode for IE -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-175311175-1"></script>
    <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-175311175-1');
    </script>
    

    <link href="/favicon.ico" rel="icon">

<link rel="canonical" href="/cv/">

        <meta name="author" content="Pierre" />
        <meta name="description" content="Associate professor at Université de Paris. A taste for challenges at the interface (human as well as scientific) and teamwork. Strong interest in computational problem solving in biology. High scientific appeal for data management and analysis of biological data. Research activities Since 2017 I am working with the Mitochondria, Metals …" />

    <meta property="og:site_name" content="Plouf" />
    <meta property="og:type" content="article"/>
    <meta property="og:title" content="CV"/>
    <meta property="og:url" content="/cv/"/>
    <meta property="og:description" content="Associate professor at Université de Paris. A taste for challenges at the interface (human as well as scientific) and teamwork. Strong interest in computational problem solving in biology. High scientific appeal for data management and analysis of biological data. Research activities Since 2017 I am working with the Mitochondria, Metals …" />



    <!-- Bootstrap -->
        <link rel="stylesheet" href="/theme/css/bootstrap.simplex.min.css" type="text/css"/>
    <link href="/theme/css/font-awesome.min.css" rel="stylesheet">

    <link href="/theme/css/pygments/default.css" rel="stylesheet">
    <link href="/theme/tipuesearch/tipuesearch.css" rel="stylesheet">
    <link rel="stylesheet" href="/theme/css/style.css" type="text/css"/>
        <link href="/static/custom.css" rel="stylesheet">

        <link href="/feeds/all.atom.xml" type="application/atom+xml" rel="alternate"
              title="Plouf ATOM Feed"/>

        <link href="/feeds/all.rss.xml" type="application/rss+xml" rel="alternate"
              title="Plouf RSS Feed"/>

</head>
<body>

<div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-ex1-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a href="/" class="navbar-brand">
Plouf            </a>
        </div>
        <div class="collapse navbar-collapse navbar-ex1-collapse">
            <ul class="nav navbar-nav">
                    <li><a href="/">Home</a></li>
                    <li><a href="/about/">About</a></li>
                    <li><a href="/research/">Research</a></li>
                    <li><a href="/python/">Python</a></li>
                    <li><a href="/tags.html">Tags</a></li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
              <li><span>
                <form class="navbar-search" action="/search.html">
                  <input type="text" class="search-query" placeholder="Search" name="q" id="tipue_search_input" required>
                </form></span>
              </li>
            </ul>
        </div>
        <!-- /.navbar-collapse -->
    </div>
</div> <!-- /.navbar -->

<!-- Banner -->
<!-- End Banner -->

<!-- Content Container -->
<div class="container">
    <div class="row">
        <div class="col-sm-9">
    <section id="content" class="body">
        <h1 class="entry-title">CV</h1>
        
        <div class="entry-content">
            <p>Associate professor at <a href="https://u-paris.fr/en/">Université de Paris</a>.</p>
<p>A taste for challenges at the interface (human as well as scientific) and teamwork. Strong interest in computational problem solving in biology. High scientific appeal for data management and analysis of biological data.</p>
<h1>Research activities</h1>
<h2>Since 2017</h2>
<p>I am working with the <a href="http://www.ijm.fr/en/research/research-groups/mitochondria/">Mitochondria, Metals and Oxidative Stress</a> research group and the <a href="http://www.ijm.fr/en/facilities/mass-spectrometry/">Proteomics &amp; mass spectrometry</a> facility, at <a href="https://www.ijm.fr/">Institut Jacques Monod</a>. </p>
<p>In tight collaboration with wet-lab biologists, I develop new methods to analyse mass spectrometry data.</p>
<p>Related publications:</p>
<ul>
<li><a href="/publications/#paper26">Novel Insights into Quantitative Proteomics from an Innovative Bottom-Up Simple Light Isotope Metabolic (bSLIM) Labeling Data Processing Strategy (2021)</a>,</li>
<li><a href="/publications/#paper25">AutoClassWrapper: a Python wrapper for AutoClass C classification (2019)</a>,</li>
<li><a href="/publications/#paper24">Pixel: a content management platform for quantitative omics data (2019)</a>.</li>
</ul>
<h2>2014-2016</h2>
<p>Visiting scientist at the Fondation Congolaise pour la Recherche Médicale (<a href="http://www.fcrm-congo.com/">FCRM</a>).</p>
<p>I am studying high-burden deseases like HIV (especially prevention of mother-to-child transmission), tuberculosis (improvement of clinical diagnosis) and malaria.</p>
<p>One of my projects concerns the development of digital solutions to fight malaria. By digital solutions I mean web or mobile applications to improve the diagnosis of malaria or the prevention and knowledge of this disease. To achieve this goal, I am working with <a href="http://fongwama.com/">Fongwama</a>, the Congolese Plateforme for open-source development, created by Fondation ECAir and <a href="http://skepsos.fr/">Skepsos</a>. </p>
<p>Related publications:</p>
<ul>
<li><a href="/publications/#paper20">HIV-1 Genetic Diversity in the Republic of Congo: Seventeen Years in Review (2015)</a>,</li>
<li><a href="/publications/#paper19">Diagnosis of smear-negative pulmonary tuberculosis based on clinical signs in the Republic of Congo (2015)</a>,</li>
<li><a href="/publications/#paper18">Factors influencing acceptability of voluntary HIV testing among pregnant women in Gamboma, Republic of Congo (2015)</a>,</li>
<li><a href="/publications/#paper17">Prevention of Mother-to-Child HIV Transmission (PMTCT) in the Republic of Congo: Challenges to Implementation (2015)</a>.</li>
</ul>
<h2>2007 - 2012</h2>
<p>Research work performed in the <a href="http://www.dsimb.inserm.fr/">DSIMB</a>
bioinformatics group of the lab “Protéines de La Membrane Erythrocytaire
et Homologues Non-Erythroides” (<a href="http://www.inserm.fr/">Inserm</a> UMR-S
665 and University Paris Diderot – Paris 7), hosted at
the <a href="http://www.ints.fr/">INTS</a>.<br>
topics: macromolecular assemblies, structural bioinformatics<br>
collaborators: 
D. Flatters (now at <a href="http://www.mti.univ-paris-diderot.fr/fr/accueil.html" title="MTi Homepage">MTi</a>),
<a href="http://www.dsimb.inserm.fr/~gelly/" title="JC Gelly Homepage">JC. Gelly</a> (DSIMB), 
C. Etchebest (DSIMB), 
<a href="http://www.dsimb.inserm.fr/~debrevern/index.php" title="A. de Brevern Homepage">A. de Brevern</a> (DSIMB), 
V. Jallu (Laboratoire d’Immunologie Plaquettaire, INTS),
<a href="http://www-lbt.ibpc.fr/people/prevost">C. Prévost</a> (Laboratoire de Biochimie Théorique, IBPC, CNRS).</p>
<p>Related publications:</p>
<ul>
<li><a href="/publications/#paper16">Protein flexibility in the light of structural alphabets (2015)</a>,</li>
<li><a href="/publications/#paper15">An Integrative Approach to the Study of Filamentous Oligomeric Assemblies, with Application to RecA (2015)</a>,</li>
<li><a href="/publications/#paper14">Modeling and molecular dynamics simulations of the V33 variant of the integrin subunit β3: Structural comparison with the L33 (HPA-1a) and P33 (HPA-1b) variants (2014)</a>,</li>
<li><a href="/publications/#paper13">Cis–trans isomerization of omega dihedrals in proteins (2013)</a>,</li>
<li><a href="/publications/#paper12">The αIIb p.Leu841Met (Cab3a+) Polymorphism Results in a New Human Platelet Alloantigen Involved in Neonatal Alloimmune Thrombocytopenia (2013)</a>,</li>
<li><a href="/publications/#paper11">Modeling and Molecular Dynamics of HPA-1a and -1b Polymorphisms: Effects on the Structure of the β3 Subunit of the αIIbβ3 Integrin (2012)</a>,</li>
<li><a href="/publications/#paper10">Ten Simple Rules for Getting Help from Online Scientific Communities (2011)</a>,</li>
<li><a href="/publications/#paper8">Detection and Architecture of Small Heat Shock Protein Monomers (2010)</a>.</li>
</ul>
<h2>2006 - 2007</h2>
<p>Post-doc at Laboratoire de Biochimie Théorique , UPR 9080 CNRS, Paris,
France.<br>
topics : flexible protein/DNA docking, coarse grain DNA model
development, C++/Python library
(<a href="http://cupnet.net/research/ptools/" title="project: PTools">PTools</a>)<br>
collaborators : A. Saladin and <a href="http://www-lbt.ibpc.fr/LBT/index.php?page=chantal-2">C.
Prévost</a></p>
<p>Publications:</p>
<ul>
<li><a href="/publications/#paper9">Modeling the early stage of DNA sequence recognition within RecA nucleoprotein filaments (2010)</a>.</li>
<li><a href="/publications/#paper7">PTools: an opensource molecular docking library (2009)</a>,</li>
<li><a href="/publications/#paper5">Insights on protein-DNA recognition by coarse grain modelling (2008)</a>,</li>
</ul>
<h2>2003 - 2006</h2>
<p>PhD student at Laboratoire de spectrométie ionique et moléculaire
(<a href="http://www-lasim.univ-lyon1.fr/">Lasim</a>) , UMR 5579, CNRS and
Université Lyon 1, France.<br>
topics : statistical approach of the structure and dynamics of isolated
proteines<br>
supervisor: <a href="http://ilm.univ-lyon1.fr/index.php?option=com_mipersonal&amp;task=2&amp;qui=311">P.
Dugourd</a>,
head of the "Electric dipole, biomolecules and clusters" group.<br>
abstract : <em>this work is a theoretical study of the thermodynamical
properties of polypeptides in the gas phase. We aim to a better
understanding of the fundamental mechanisms involved in protein folding.
A statistical approach based on Monte Carlo algorithms applied in
generalized ensembles, such as Replica Exchange Method or Wang-Landau
method, has been used to sample the rugged energy landscape of these
molecules. The peptides were composed of 2 to 20 amino acids and have
been modelized by the Amber 96 forcefield. Simulations have been
realized in strong relation with experimental progress of the group. We
thus tried to understand the influence of the secondary structure on
photofragmentation mechanism, the role of entropy in the stabilisation
of beta sheets and the effect of intense and static electric field on
peptide conformation.</em><br>
Download my PhD thesis locally <a href="http://pierrepoulain.free.fr/pdf/these-poulain.pdf">[PDF, 4.9 Mb,
French]</a> or on
the <a href="http://tel.archives-ouvertes.fr/tel-00094458">CNRS TEL</a>
multidisciplinary thesis server.</p>
<p>Publications:</p>
<ul>
<li><a href="/publications/#paper6">Transitions between secondary structures in isolated polyalanines (2009)</a>,</li>
<li><a href="/publications/#paper4">Competition between secondary structures in gas phase polyalanines (2007)</a>,</li>
<li><a href="/publications/#paper3">Performances of Wang-Landau algorithms for continuous systems (2006)</a>,</li>
<li><a href="/publications/#paper2">Comparison of the fragmentation pattern induced by collisions, laser excitation and electron capture. Influence of the initial excitation (2006)</a>,</li>
<li><a href="/publications/#paper1">Monte Carlo simulations of flexible molecules in a static electric field: electric dipole and conformation (2005)</a>,</li>
</ul>
<h2>Older &amp; shorter</h2>
<h3>March - September 2003</h3>
<p>Biomolecular Sciences department, UMIST, Manchester, UK<br>
topics: automatic extraction of the toxic properties of compounds from
the biomedical literature<br>
supervisor: B. J. Stapley<br>
abstract: <em>Automatic information extraction from free-text documents
encompasse a set of techniques and tools already well known, but its
application to biology is relatively new. Thus, this project aims to
predict and/or extract toxicological properties of some compounds using
information extraction and retrieval techniques. Nowadays, toxicological
data are issued from in vitro and in vivo tests that are expensive and
time-consuming for the pharmaceutical industry. As a consequence, and
given the large body of information latent in bibliographic databases,
the prediction of toxic information on possible therapeutic compounds
becomes particularly important.</em></p>
<h2>July - August 2001</h2>
<p>Laboratoire d'ImmunoGénétique Moléculaire (LIGM) , Montpellier,
France.<br>
topics: 3D visualization tools for IMGT structural data<br>
supervisors: M. Ruiz and <a href="http://www.imgt.org/IMGTinformation/CVMPL.html">M.-P.
Lefranc</a><br>
abstract: <em><a href="http://www.imgt.org/">IMGT</a>, the international
ImMunoGeneTics database is an integrated database, specializing in
immunoglobulins (Ig), T-cells receptors (TcR) and major histocompatibily
complex (MHC) molecules of all species that was created in 1989 by
Marie-Paule Lefranc (Montpellier II Unversity, CNRS, Montpellier,
France) We aim to implement PERL scripts in order to compare 3D
structures of proteins picked up in the IMGT database (use of STAMP).</em></p>
<h1>Teaching activities</h1>
<p>High involvement in pedagogy:</p>
<ul>
<li>Certificate in pedagogy in 2019 (Certifiens).</li>
<li>Awards in digital educational innovations: EdTech 2018 for the <a href="https://plasmabio.org/">Plasma project</a>, IN2018 (live coding), Idex2020 (flipped classroom).</li>
</ul>
<p><strong>2007 - to present</strong><br>
Université de Paris.<br>
Python programming, unix, proteomics, open science, bioinformatics project management (data, source code...). <br>
Creation of two professionnal trainings in bioinformatics and omics data analysis.</p>
<p><strong>2003 - 2006</strong><br>
Multidisciplanary engineering university Institut National des Sciences
Appliquées de Lyon (INSA Lyon)<br>
First year chemistry workshops (total of 288 hours, 255 students).</p>
<p><strong>22 - 27/05/2005</strong><br>
Summer school "Modelisation, dynamique et thermodynamique des agrégats
et molécules complexes"<br>
Modelisation workshop: simulated annealing and parallel tempering
simulations of a dipeptide.</p>
<h1>Collective duty</h1>
<p>Member of the steering committee of :</p>
<ul>
<li>the integrative platform for omics projects at Université de Paris (iPOP-UP)</li>
<li>the bioinformatics and biostatistics platform of Epigenetics and Cell Fate lab</li>
<li>IT departement of Institut Jacques Monod</li>
</ul>
<p>Head of the IT service of the Life Sciences Department</p>
<h1>Academic training</h1>
<p><strong>2003 - 2006</strong><br>
PhD in Physics, <a href="http://www-lasim.univ-lyon1.fr/">Lasim</a>, Université
Lyon 1, France</p>
<p><strong>2002 - 2003</strong><br>
MSc in Cheminformatics, UMIST, Manchester, U.K.</p>
<p><strong>2000 - 2003</strong><br>
"<a href="https://www.enscm.fr/en/cursus/engineer-degree/">Diplôme d'Ingénieur Chimiste</a>" at the <a href="http://www.enscm.fr/">Ecole Nationale Supérieure de
Chimie de Montpellier</a> (the National Graduate
School of Chemistry, Montpellier)  </p>
<p><strong>1997 - 2000</strong><br>
Classes Préparatoires aux Grandes Ecoles, preparatory classes leading
to French institutes. Lycée Henri Bergson, Angers, France.</p>
<p><strong>1997</strong><br>
Baccalauréat in Sciences, i.e. High School diploma<br>
Lycée David d'Angers, France.</p>
<h1>Languages</h1>
<ul>
<li>French: native fluency</li>
<li>English: good level in spoken and written English</li>
<li>Munukutuba: elementary proficiency</li>
</ul>
<h1>Computers skills</h1>
<ul>
<li>operating systems [administration]: GNU/Linux (Ubuntu, Debian)</li>
<li>office software: Microsoft Office and OpenOffice.org</li>
<li>programming: Python, Bash</li>
<li>data analysis and visualisation with Python: Pandas, Matplotlib, Plotly</li>
<li>text formating: LaTeX, HTML, Markdown</li>
</ul>
<h1>Professional address</h1>
<p><a href="http://www.ijm.fr/en/">Institut Jacques Monod</a> (IJM)<br>
Building Buffon<br>
15 rue Hélène Brion<br>
75015 Paris
France</p>
<p>My office is on the third floor, room 390B. Here are <a href="http://www.ijm.fr/en/access/" title="How to get to IJM?">directions</a> for visitors to find the IJM.</p>
        </div>
    </section>
        </div>
        <div class="col-sm-3" id="sidebar">
            <aside>
<!-- Sidebar -->
<section class="well well-sm">
  <ul class="list-group list-group-flush">

<!-- Sidebar/Social -->
<li class="list-group-item">
  <h4><i class="fa fa-home fa-lg"></i><span class="icon-label">Social</span></h4>
  <ul class="list-group" id="social">
    <li class="list-group-item"><a href="https://twitter.com/pierrepo"><i class="fa fa-twitter-square fa-lg"></i> Twitter</a></li>
    <li class="list-group-item"><a href="http://fr.linkedin.com/in/pierrepo"><i class="fa fa-linkedin-square fa-lg"></i> LinkedIn</a></li>
    <li class="list-group-item"><a href="https://github.com/pierrepo"><i class="fa fa-github-square fa-lg"></i> GitHub</a></li>
    <li class="list-group-item"><a href="http://cupnet.net/feed/all.rss.xml"><i class="fa fa-rss-square fa-lg"></i> rss</a></li>
  </ul>
</li>
<!-- End Sidebar/Social -->

<!-- Sidebar/Links -->
<li class="list-group-item">
  <h4><i class="fa fa-external-link-square fa-lg"></i><span class="icon-label">Links</span></h4>
  <ul class="list-group" id="links">
    <li class="list-group-item">
      <a href="https://github.com/pierrepo" target="_blank"><i class="fa fa-github fa-lg"></i> GitHub</a>
    </li>
    <li class="list-group-item">
      <a href="https://orcid.org/0000-0003-4177-3619" target="_blank"><i class="fa fa-user fa-lg"></i> ORCID</a>
    </li>
    <li class="list-group-item">
      <a href="https://dissem.in/r/1540/pierre-poulain/" target="_blank"><i class="fa fa-sticky-note fa-lg"></i> Dissemin</a>
    </li>
  </ul>
</li>
<!-- End Sidebar/Links -->
  </ul>
</section>
<!-- End Sidebar -->            </aside>
        </div>
    </div>
</div>
<!-- End Content Container -->

<footer>
   <div class="container">
      <hr>
      <div class="row">
         <div class="col-xs-10">&copy; 2020 Pierre Poulain
            &middot; Powered by <a href="https://github.com/getpelican/pelican-themes/tree/master/pelican-bootstrap3" target="_blank">pelican-bootstrap3</a>,
            <a href="http://docs.getpelican.com/" target="_blank">Pelican</a>,
            <a href="http://getbootstrap.com" target="_blank">Bootstrap</a>                <p><small>  <a rel="license" href="https://creativecommons.org/licenses/by/4.0/deed.en"><img alt="Creative Commons License" style="border-width:0" src="//i.creativecommons.org/l/by/4.0/80x15.png" /></a>
    Content
  licensed under a <a rel="license" href="https://creativecommons.org/licenses/by/4.0/deed.en">Creative Commons Attribution 4.0 International License</a>, except where indicated otherwise.
</small></p>
         </div>
         <div class="col-xs-2"><p class="pull-right"><i class="fa fa-arrow-up"></i> <a href="#">Back to top</a></p></div>
      </div>
   </div>
</footer>
<script src="/theme/js/jquery.min.js"></script>

<!-- Include all compiled plugins (below), or include individual files as needed -->
<script src="/theme/js/bootstrap.min.js"></script>

<!-- Enable responsive features in IE8 with Respond.js (https://github.com/scottjehl/Respond) -->
<script src="/theme/js/respond.min.js"></script>





</body>
</html>